### DISCUSSION TITLE
Error with cron job?

### DISCUSSION THREAD

**Discussion by mpaglia0** [mosparo/mosparo#173] [Problems & Issues]
When `bin/console mosparo:ruleset-refresh` runs I receive via email the following message:

In Input.php line 195:
                                                                               
  Attempted to call function "escapeshellarg" from namespace "Symfony\Compone  
  nt\Console\Input".

**Comment by zepich** [comment]
Hi @mpaglia0 

Since `escapeshellarg` is a standard functionality and should always be available, I guess it is because the function is disabled on your hosting. Do you use shared hosting (cPanel, Plesk, or something like that), and do you have an option to adjust the disabled functions? If yes, please remove the `escapeshellarg` from the list there.

If you do not have shared hosting, please tell me your PHP version.

Thank you very much for your answer.

Kind regards,

zepich

**Reply by mpaglia0** [reply]
Ciao @zepich 
I cannot find `escapeshellarg` among features I can enable/disable in my PHP dashboard.
For your information I am using PHP 8.1

Thank you,
Maurizio

**Reply by zepich** [reply]
Hi Maurizio

What kind of control panel do you have? Plesk or cPanel or something else?

Can you check what you see in the PHP information? If you use `php -i` (when you can access the server by SSH) or `<?php phpinfo(); ?>` in a PHP test file, you should see all the configuration values from PHP. One of these is `disable_functions`. Can you check what value you see there?

Kind regards,

zepich

**Reply by mpaglia0** [reply]
Ciao @zepich 
I have cPanel installed n my hosting service.
Value of `disable_function` parameter are the following:
system,passthru,exec,popen,proc_close,proc_get_status,proc_nice,proc_open,proc_terminate,shell_exec,escapeshellcmd,define_syslog_variables,posix_uname,posix_getpwuid,apache_child_terminate,posix_kill,posix_mkfifo,posix_setpgid,posix_setsid,posix_setuid,escapeshellarg,posix_uname,ftp_exec,ini_alter,ini_restore,inject_code,syslog,openlog,define_syslog_variables,apache_setenv,eval,phpAds_XmlRpc,phpAds_remoteInfo,phpAds_xmlrpcEncode,phpAds_xmlrpcDecode,xmlrpc_entity_decode,fp,fput

**Reply by zepich** [reply]
Hi Maurizio

Thank you very much for your update.

Okay, so the function is on the list of disabled functions, and that's why it's not working correctly.

I guess your hoster put the function on this list because it's generally not a safe functionality. There is no risk for our use case since Symfony uses it for the CLI commands, but the hoster will block them because they're not needed for the web (I guess).

Unfortunately, I have no solution for this problem right now.

You could ask your hoster to remove it from the list of disabled functions, at least for the CLI. But I guess the hoster does not want to do that (which is understandable).

I will add a task for this to our list of ideas to add a command to execute all these cronjobs by requesting a URL. Your hoster is probably not the only one with this function disabled and we aim to make mosparo usable on (more or less) every hosting. Until this solution is added, it will take some time since we will add this in v1.2 (and the release of v1.1 is planned for next week, so I guess the release of v1.2 will be around April/May).

The good thing is that mosparo does not directly need the cronjobs for the database cleanup and the ruleset download since mosparo also executes these tasks in the normal process. The process (for the validation, for example) will take a bit longer, but it will download the ruleset. The only task that mosparo is not doing automatically is downloading the GeoIP2 database. You have to do that with the cronjob or the button in the administration interface.

I'm sorry for this issue.

Kind regards,

zepich

**Reply by zepich** [reply]
(See #174)

**Reply by mpaglia0** [reply]
Thank you indeed for your kindness and availability.

Have a nice day,
Maurizio
